Added common terms section to the Installing Book (#46421)
openshift/openshift-docs: modules/install-openshift-common-terms.adoc

// Module included in the following assemblies:
//
// * installing/index.adoc

:_content-type: REFERENCE
[id="install-openshift-common-terms_{context}"]
= Common terms

This glossary defines common terms that are used in the installation content. These terms help you understand installation effectively.

Bootstrap node::

> A temporary machine that runs a minimal Kubernetes configuration to deploy the {product-title} control plane. It is deleted at the end of the installation.

Reviewer: > It is deleted at the end of the installation.
for IPI, it will be deleted after bootstrap process successfully complete.
for UPI, it could be safely deleted after bootstrap process successfully complete by user.

Author: Removed the second sentence to give clarity

Control plane::

>  It is a container orchestration layer that exposes the API and interfaces to define, deploy, and manage the lifecycle of containers. Also known as control plane machines.

Reviewer 2: I think you can omit "Is is a" to align with the structure you used in the Bootstrap node definition

>  It is a container orchestration layer that exposes the API and interfaces to define,deploy, and manage the lifecycle of containers. Also known as control plane machines.

Reviewer: `define,deploy` -> `define, deploy` ?

Author: please elaborate what you mean ??

Author: There is a similar definition in the Kubernetes glossary >> https://kubernetes.io/docs/reference/glossary/?fundamental=true
@yunjiang29

Reviewer: @Amrita42 I mean a **space** is missing between `define,` and `deploy`, sorry for confused info.

Compute node::
Nodes that are responsible for executing workloads for cluster users. Also known as worker nodes.

Disconnected installation::
There are situations where parts of a data center might not have access to the internet, even through proxy servers. You can still install the{product-title} in these environments, but you must download the required software and images and make them available to the disconnected environment.

The {product-title} installation program::
A program that provisions the infrastructure and deploys a cluster.

Installer-provisioned infrastructure::
The installation program deploys and configures the infrastructure that the cluster runs on.

Ignition config files::

> A file that Ignition uses to configure Red Hat Enterprise Linux CoreOS (RHCOS) during operating system initialization. The installation program generates different Ignition config files to initialize bootstrap, control plane, and worker nodes.

Reviewer: > The installation program generates different Ignition config files to initialize bootstrap, control plane, and worker nodes.

I'm not sure if this description is accurate, b/c:
* The control plane, and worker nodes' ignitions are not _files_ but _endpoints_ (a URL). 
* Are control plane, and worker nodes' ignitions are generated by installer ? please @patrickdillon help to confirm,  thanks.

Reviewer 3: The installer does create ignition config files for bootstrap, control plane and compute nodes. You are correct that the files for control plane and compute are stub ignition configs that only contain endpoints. I think the current wording of the pr is fine, especially for an introductory audience this is intended for.

Kubernetes manifests::
Specifications of a Kubernetes API object in a JSON or YAML format. A configuration file can include deployments, config maps, secrets, daemonsets etc.

Kubelet::

> A **primary node** agent that runs on each node in the cluster to ensure  that containers are running in a pod.

Reviewer 4: Remove extra space in "...to ensure  that".

Load balancers::

> A service to allocate a unique IP from a configured pool.  It  has a single edge router IP, which can be a virtual IP (VIP), but is still a single service for initial load balancing.

Reviewer 3: This definition reads like the k8s definition of a load balancer, which is not really what we use for installs (at least not yet, unfortunately). 

In the context of an install, I would suggest a definition such as this, which I cribbed from aws:
> A load balancer serves as the single point of contact for clients. Load balancers for the API distribute incoming  traffic across control-plane nodes.

Machine Config Operator::

> An operator that manages and applies configuration and updates of the base operating system and container runtime, including everything between the kernel and kubelet.

Reviewer 3: ```suggestion
An operator that manages and applies configuration and updates of the base operating system and container runtime, including everything between the kernel and kubelet, for nodes in the cluster.
```

Operators::

> The preferred method of packaging, deploying, and managing services on the control plane.

Reviewer 3: I'm not sure if "on the control plane" is accurate here... These [two](https://operatorframework.io/) [pages](https://operatorframework.io/about/) from the Operator Framework have good descriptions. 

If you want a simple fix I would just do:
```suggestion
The preferred method of packaging, deploying, and managing services in an OpenShift cluster.
```

but I do think it would be beneficial to flesh this out a bit.

Author: @patrickdillon I have re-phrased. Could you please have a look, and if all looks good . ack it.

> The preferred method of packaging, deploying and managing a Kubernetes application in an {product-title} cluster. It takes human operational knowledge and encodes it into software that is easily packaged and shared with customers.

Reviewer 2: deploying, and managing  (Oxford comma)

An Operator takes human operational knowledge

User-provisioned infrastructure::

> You can install {product-title} on infrastructure that you provide. You can use the installation program to generate the assets that you require to provision the cluster infrastructure, create the cluster infrastructure, and then deploy the cluster to the infrastructure that you provided.

Reviewer 4: Rephrase as: "...to generate the assets requires to..."

// Module included in the following assemblies:
//
// * installing/index.adoc

:_content-type: REFERENCE
[id="install-openshift-common-terms_{context}"]
= Common terms

This glossary defines common terms that are used in the installation content. These terms help you understand installation effectively.

Control plane::

> Th control plane manages the worker nodes and the pods in the {product-title} cluster. It is a container orchestration layer that exposes the API and interfaces to define,deploy, and manage the lifecycle of containers.

Reviewer 5: That doesn't sound very accurate for me. In IPI, it should be machine-api that manages the worker nodes, in UPI, the user manages the worker nodes.

Reviewer 5: I found there's already some comprehensive description on `Control planes` in https://docs.openshift.com/container-platform/4.10/architecture/control-plane.html#defining-masters_control-plane

// re-written from ocp repo glossary

Compute node::
Compute nodes also known as worker nodes, are nodes which are responsible for executing workloads for the cluster users.
//re-written
Bootstrap node::
A bootstrap node is a temporary machine that runs a minimal Kubernetes configuration that is used to deploy the {product-title} control plane. It is deleted at the end of the installation.
//re-written

Machine Config Operator::
The Machine Config Operator manages and applies configuration and updates of the base operating system and container runtime, including everything between the kernel and kubelet.
//from OCP docs

Ignition config files::
The file that Ignition uses to configure Red Hat Enterprise Linux CoreOS (RHCOS) during operating system initialization. The installation program generates different Ignition config files to initialize bootstrap, control plane, and worker nodes.
//RH docs

Kubernetes manifests::
Kubernetes manifests are the specifications of a Kubernetes API object in a JSON or YAML format. A configuration file can include deployments, config maps, secrets, daemonsets etc.
//re-written

Load balancers::

> A load balancer service allocates a unique IP from a configured pool. The load balancer has a single edge router IP, which can be a virtual IP (VIP), but is still a single machine for initial load balancing).

Reviewer 5: extra ")" in the end.

//ocp docs

Kubelet::
A **primary node** agent that runs on each node in the cluster. It ensures that containers are running in a pod.
//re-written

Operators::
Operators are the preferred method of packaging, deploying, and managing services on the control plane.
//OCP docs
Disconnected installation::

> There are situations where parts of a data center might not have access to the internet, even through proxy servers. You can still install {product-title} in these environments, but you must download the required software and images and make them available to the disconnected environment.

Reviewer 5: Better to use `restricted network installation` to align with our current installation guide, such as:
https://docs.openshift.com/container-platform/4.10/installing/installing_aws/installing-restricted-networks-aws-installer-provisioned.html

Author: I would suggest disconnected  e.g. https://docs.openshift.com/container-platform/4.10/installing/disconnected_install/installing-mirroring-creating-registry.html. There is an open PR to standardize on the terminology usage >> https://issues.redhat.com/browse/OSDOCS-3685

//re-written

> The {product-title} installation program::

Reviewer 5: Duplicated with the above `Installation program ` item.
